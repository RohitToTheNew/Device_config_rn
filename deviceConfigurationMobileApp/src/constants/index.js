import UUIDMappingMS700 from './uuidMapping-ms700';
import UUIDMappingInfoView from './uuidMapping-infoview';
import DefaultCredentials from './defaultCredentials';
const boudRate = [9600, 19200, 38400, 57600, 115200]

export {DefaultCredentials, UUIDMappingMS700, UUIDMappingInfoView, boudRate};
