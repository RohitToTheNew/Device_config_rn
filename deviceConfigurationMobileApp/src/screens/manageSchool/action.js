const identifire = 'manage-school-log: ';
import { Alert } from 'react-native';
import RNFS from 'react-native-fs';
import Mailer from 'react-native-mail';
import Utils from '../../utils';
import { translate } from '../../translations/translationHelper';

/**
 * function to get saved Schools data from local storage
 */
function getSavedData() {
  const result = RNFS.readFile(RNFS.DocumentDirectoryPath + '/test.txt', 'utf8');
  return result;
}

/**
 * function for formatting the rooms data to desired CSV format
 * @param {object} parms object containing the data of rooms
 * @returns formatted CSV data
 */
function getDataInDesiredFormate(parms) {
  let CSVData = [];
  parms.forEach(room => {
    for (const key in room) {
      if (key === 'devices') {
        room[key].forEach(roomData => {
          let obj = {};
          obj.driverName = roomData.deviceType;
          obj.deviceName = roomData.deviceName;
          obj.ip = roomData.ipAddress || '';
          obj.mac = roomData.macAddress;
          obj.roomName = room['title'];
          CSVData.push(obj);
        });
      }
    }
  });
  return CSVData;
}

/**
 * function to save schools list to local storage
 * @param {object} schools object containing schools list
 * @returns 
 */
const writeDataTolocalStorage = async schools => {
  const data = JSON.stringify(schools);
  const path = RNFS.DocumentDirectoryPath + '/test.txt';
  try {
    await RNFS.writeFile(path, data, 'utf8');
    return `FILE WRITTEN!`;
  } catch (err) {
    Utils.Log(Utils.logType.error, 'error in writeDataTolocalStorage', err)
    return `Error while writing`;
  }
};

/**
 * function to write schools data to local storage
 * @param {object} schools object containing data of schools
 * @param {string} fileName name of file containing data of schools
 * @returns error(if any) during wiriting to local storage OR success message if writing is successful
 */

const writeCSVDataTolocalStorage = async (schools, fileName) => {
  const resultCSV = convertToCSV(schools);
  const pathToWrite = `${RNFS.DocumentDirectoryPath}/${fileName}.csv`;
  try {
    await RNFS.writeFile(pathToWrite, resultCSV, 'utf8');
    return `FILE WRITTEN!`;
  } catch (err) {
    Utils.showToast(translate('errorWhileWriting.'));
    return `Error while writing`;
  }
};
/**
 * function to attach generated CSV in email and open email client
 * @param {string} fileName name of file containing data of schools
 */
function sendCSVAsMail(schoolName, fileName) {
  try {
    Mailer.mail(
      {
        subject: `${schoolName} Devices`,
        isHTML: true,
        attachments: [
          {
            type: 'csv',
            path: `${RNFS.DocumentDirectoryPath}/${fileName}.csv`, // The absolute path of the file from which to read data.
          },
        ],
      },

      (error, event) => {
        Alert.alert(
          error,
          event,
          [
            {
              text: 'Ok',
              onPress: () => { },
            },
            {
              text: 'Cancel',
              onPress: () => { },
            },
          ],
          { cancelable: true },
        );
      },
    );
  } catch (error) {
    Utils.Log(Utils.logType.error, 'error in sendCSVAsMail', error);
  }
}

/**
 * function to handle the 'Export CSV' action for school
 * @param {object} parms object containig data of school
 */
async function handleExportCSV(parms) {
  try {
    const schoolName = parms?.schoolTitle;

    let month = new Date().getMonth() + 1;
    let date = new Date().getDate();
    const year = new Date().getFullYear();

    if (date < 10) date = '0' + date;
    if (month < 10) month = '0' + month;

    const hour = new Date().getHours();
    const minute = new Date().getMinutes();
    const second = new Date().getSeconds();

    const currentDate = `${month}-${date}-${year}`;
    const currentTime = `${hour}H${minute}M${second}S`;

    const fileName = `${schoolName}_Devices_${currentDate}_${currentTime}`
      .toString()
      .trim();
    const CSVData = getDataInDesiredFormate(parms?.rooms);
    await writeCSVDataTolocalStorage(CSVData, fileName);
    sendCSVAsMail(schoolName, fileName);
  } catch (error) {
    Utils.Log(Utils.logType.error, 'handleExportCSV', error);
  }
}

/**
 * function to convert the object data to desired CSV format
 * @param {object} arr array of rooms details
 * @returns
 */
function convertToCSV(arr) {
  const array = [Object.keys(arr[0])].concat(arr);
  return array
    .map(it => {
      return Object.values(it).toString();
    })
    .join('\n');
}

/**
 * function to show alert when delete button is pressed
 * @param {string} schoolTitle title of school
 * @param {string} schoolId school id
 * @param {function} setSchoolTitle function to set school title
 * @param {function} setSchoolId function to set school id
 * @param {function} setPopupToDeleteDevice function to set popup to show
 */
function deleteSchoolAlert(
  schoolTitle,
  schoolId,
  setSchoolTitle,
  setSchoolId,
  setPopupToDeleteDevice,
) {
  setSchoolTitle(schoolTitle);
  setSchoolId(schoolId);
  setPopupToDeleteDevice(true);
}

/**
 * function to delete school
 * @param {string} schoolID school id
 * @param {object} savedData object containing saved schools data
 * @param {function} setSavedData function to set saved schools data after change
 * @param {*} setPopupToDeleteDevice function to set popup to hide
 */
async function deleteSchool(
  schoolID,
  savedData,
  setSavedData,
  setPopupToDeleteDevice,
) {
  try {
    const filteredData = savedData.filter(school => school?.id !== schoolID);
    await writeDataTolocalStorage(filteredData);
    setSavedData(filteredData);
    setPopupToDeleteDevice(false);
  } catch (error) {
    Utils.Log(Utils.logType.error, 'deleteSchool', error);
  }
}
export {
  getSavedData,
  getDataInDesiredFormate,
  writeCSVDataTolocalStorage,
  sendCSVAsMail,
  handleExportCSV,
  deleteSchoolAlert,
  deleteSchool,
  writeDataTolocalStorage,
};
