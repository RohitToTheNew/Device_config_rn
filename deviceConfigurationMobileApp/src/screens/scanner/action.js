const identifire = 'scanner-action log: ';
import { check, request, RESULTS, PERMISSIONS } from 'react-native-permissions';
import { RNCamera } from 'react-native-camera';
import utils from '../../utils';
import { translate } from '../../translations/translationHelper';
import RNFS from 'react-native-fs';

/**
 * On click Enter details manually and redirect user to Add device screen
 * @param {object} navigation it represents navigation object
 * @returns object
 */
function actionHandleManualDetails(navigation) {
  return () => {
    navigation.navigate('addDevice');
  };
}

/**
 * function to read local file data from storage
 * @returns file details read from storage
 */
const readFileFromLocalStorage = async () => {
  let result = RNFS.readFile(RNFS.DocumentDirectoryPath + '/test.txt', 'utf8');
  return result;
};

/**
 * function to check if the device details encoded into the QR code is having valid keys
 * @param {object} obj object containing the scanned data from QR code
 * @returns boolean indicating weather details are valid or not
 */
const isValidObject = (obj) =>
  obj?.serialNumber && obj?.serialNumber !== '' &&
  obj?.macAddress && obj?.macAddress !== '' &&
  obj?.deviceType && obj?.deviceType !== '';

/**
 * function to check if the device details encoded into the QR code is having valid keys
 * @param {object} obj object containing the scanned data from QR code
 * @returns boolean indicating weather details are valid or not
 */
const isValidDetails = (obj) =>
  obj?.serial && obj?.serial !== '' &&
  obj?.mac && obj?.mac !== '' &&
  obj?.device && obj?.device !== '';

/**
 * On successful scan
 * @param {object} navigation it represents navigation object
 * @param {function} setInvalidCode used to set invalid code boolean value
 * @param {object} e contains scan result
 */
function actionOnSuccess(navigation, setInvalidCode, e) {
  const typeToCheck = utils.isIOS ? 'org.iso.QRCode' : "QR_CODE"
  if (e.type === typeToCheck) {
    try {
      const scannedString = JSON.parse(e?.data);
      console.log('scanned string', scannedString)
      if (isValidObject(scannedString) || isValidDetails(scannedString)) {
        navigation.replace('addDevice', {
          mac: scannedString?.macAddress || scannedString?.mac,
          serialNumber: scannedString?.serialNumber || scannedString?.serial,
          deviceType: scannedString?.deviceType || scannedString?.device,
        });
      } else {
        setInvalidCode(true);
      }
    } catch (e) {
      setInvalidCode(true);
    }
  } else {
    setInvalidCode(true);
  }
}

/**
 * Click on close icon
 * @param {object} navigation it represents navigation object
 * @returns object
 */
function actionHandleClose(navigation) {
  return () => {
    navigation.goBack();
  };
}

/**
 * Click on flash icon
 * @param {boolean} isFlashOn used to identify if flash is on/off
 * @param {function} setFlash used to set flash on/off value
 * @param {function} setIsFlashOn used to set flash on value
 * @returns object
 */
function actionHandleFlash(isFlashOn, setFlash, setIsFlashOn) {
  return () => {
    if (isFlashOn) {
      setFlash(RNCamera.Constants.FlashMode.off);
      setIsFlashOn(false);
    } else {
      setFlash(RNCamera.Constants.FlashMode.torch);
      setIsFlashOn(true);
    }
  };
}

/**
 * To ask camera permission
 * @returns camera permission
 */
async function actionUseEffect() {

  try {
    const result = await check(PERMISSIONS.IOS.CAMERA);
    if (result !== RESULTS.GRANTED) {
      const askCameraPermission = await request(PERMISSIONS.IOS.CAMERA);
      return askCameraPermission;
    } else {
      utils.Log(utils.logType.log, identifire, 'Camera permisson granted');
    }
  } catch (Ex) {
    utils.showToast(translate('error'), translate('permissionsException'));
  }
}
export {
  actionHandleManualDetails,
  actionOnSuccess,
  actionHandleClose,
  actionUseEffect,
  actionHandleFlash,
};
