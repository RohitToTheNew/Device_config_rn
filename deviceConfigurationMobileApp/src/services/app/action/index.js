import BleManager from '../../../config/bleManagerInstance';
import { UUIDMappingMS700 } from '../../../constants';
import Utils from '../../../utils';
import {UPDATE_APP_MODAL_FIELDS} from '../constants';
import {sentryErrorHandler} from '../../../utils/errorHandler';
import { translate } from '../../../translations/translationHelper';

export const updateAppModalFields = (key, value) => dispatch => {
  dispatch({
    type: UPDATE_APP_MODAL_FIELDS,
    payload: {[key]: value},
  });
};

/**
 * function that reads POR override value from BLE
 */
export const readPoeOverride = () => {
  return async (dispatch, getState) => {
    const {connectedDevice} = getState().authDevices;
    try {
      const poeValue = await BleManager.readCharacteristicForDevice(
        connectedDevice.id,
        UUIDMappingMS700.rootServiceUDID,
        UUIDMappingMS700.poeOverrride,
      );
      if (Utils.decodeBase64(poeValue.value) == '\u0000\u0000\u0000' || Utils.decodeBase64(poeValue.value) == "" ) {
        dispatch(updateAppModalFields('poeOverride', false)) 
      }else{
        const valueObtained =  Utils.decodeBase64(poeValue.value) == 'true' ? true : false;
        dispatch(updateAppModalFields('poeOverride', valueObtained)) 
      }
    } catch (error) {
      Utils.Log(Utils.logType.error, 'readPoeOverride', error);
      Utils.showToast(translate('unableToReadValuse'));
    }
  };
};

/**
 * redux action that saves POE override settings onto MS700 Device
 * @param {function} callback callback function that gets executed after settings are saved
 */
export const togglePoeOverride = callback => {
  return async (dispatch, getState) => {
    const {connectedDevice} = getState().authDevices;
    const {poeOverride} = getState().app;
    try {
      const valueToUpdate = poeOverride ? "false" : "true"
      await BleManager.writeCharacteristicWithResponseForDevice(
        connectedDevice.id,
        UUIDMappingMS700.rootServiceUDID,
        UUIDMappingMS700.poeOverrride,
        Utils.encodeBase64(valueToUpdate),
      );
      await BleManager.writeCharacteristicWithResponseForDevice(
        connectedDevice.id,
        UUIDMappingMS700.rootServiceUDID,
        UUIDMappingMS700.applyChange,
        Utils.encodeBase64('')
      );
      Utils.showToast('',poeOverride ?'PoE Override disabled' :'PoE Override enabled','success')
      dispatch(updateAppModalFields('poeOverride', !poeOverride))
      callback();
    } catch (error) {
      sentryErrorHandler(`togglePoeOverride eror ${JSON.stringify(error)}`)
      Utils.showToast('Error', 'Not able to change the poe settings, try again.');
    }
  };
};