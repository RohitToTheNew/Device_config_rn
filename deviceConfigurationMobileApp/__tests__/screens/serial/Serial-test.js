import React from 'react';
import base64 from 'react-native-base64';
import renderer from 'react-test-renderer';
import Toast from 'react-native-toast-message';
import Serial from '../../../src/screens/serial';
import {useDispatch, useSelector} from 'react-redux';
import {store} from '../../../src/store/configureStore';
import {UUIDMappingMS700} from '../../../src/constants';
import BleManager from '../../../src/config/bleManagerInstance';
import {updateAuthDevices} from '../../../src/services/authDevices/action';
import * as SerialSettingsFunction from '../../../src/services/serial/action';
import {fireEvent, render} from '@testing-library/react-native';
import {NavigationContainer} from '@react-navigation/native';

jest.mock('react-redux', () => ({
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}));

describe('on homescreen bottom tab mount', () => {
  let tree, readSpy, writeSpy, toastSpy;
  const mockDispatch = jest.fn();
  const mockConnectedDevice = {
    localName: 'Brx-MockDevice',
    deviceType: 'MockDeviceType',
    id: 1,
  };
  const mockSettingsMap = [
    {
      portName: 'Remote Port 1',
      boudRate: '',
      forwardingBehaviour: [
        {portName: 'No Forward', selected: false, portNumber: 0},
        {portName: 'Remote Port 2', selected: false, portNumber: 2},
        {portName: 'Remote Port 3', selected: false, portNumber: 3},
        {portName: 'XD', selected: false, portNumber: 4},
        {portName: 'Serial', selected: false, portNumber: 5},
        {portName: 'Network/EPIC', selected: false, portNumber: 6},
      ],
      selectedBoudRate: '',
    },
  ];

  const navigation = {
    navigate: jest.fn(),
  };

  beforeAll(async () => {
    useSelector.mockImplementation(selector => {
      const initialState = {
        authDevices: {connectedDevice: mockConnectedDevice},
        app: {showModal: false, deviceType: 'MS-700'},
        serial: {settingsMap: mockSettingsMap},
      };
      if (selector) {
        return selector(initialState);
      }
    });
    useDispatch.mockReturnValue(mockDispatch);
    await store.dispatch(
      updateAuthDevices('connectedDevice', mockConnectedDevice),
    );
  });

  it('should match the snapshot', () => {
    tree = renderer
      .create(
        <NavigationContainer>
          <Serial />
        </NavigationContainer>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render the component correctly', () => {
    const {getByTestId, getByText} = render(
      <NavigationContainer>
        <Serial navigation={navigation} />
      </NavigationContainer>,
    );
    expect(getByTestId('serialComponent')).toBeDefined();
    expect(getByText('Brx-MockDevice')).toBeTruthy();
    expect(getByTestId('serialSettingsList')).toBeTruthy();
  });

  it('should fetch serial settings from MS700', async () => {
    readSpy = jest.spyOn(BleManager, 'readCharacteristicForDevice');
    await store.dispatch(SerialSettingsFunction.getSerialSettings());
    expect(readSpy).toBeCalledWith(
      mockConnectedDevice.id,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.fwBehaviourRP1,
    );
    expect(readSpy).toBeCalledWith(
      mockConnectedDevice.id,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.fwBehaviourRP2,
    );
    expect(readSpy).toBeCalledWith(
      mockConnectedDevice.id,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.fwBehaviourRP3,
    );
    expect(readSpy).toBeCalledWith(
      mockConnectedDevice.id,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.fwBehaviourSerial,
    );
    expect(readSpy).toBeCalledWith(
      mockConnectedDevice.id,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.fwBehaviourTCP,
    );
    expect(readSpy).toBeCalledWith(
      mockConnectedDevice.id,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.fwBehaviourXD,
    );
    expect(readSpy).toBeCalledWith(
      mockConnectedDevice.id,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.boudRateRP1,
    );
    expect(readSpy).toBeCalledWith(
      mockConnectedDevice.id,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.boudRateRP2,
    );
    expect(readSpy).toBeCalledWith(
      mockConnectedDevice.id,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.boudRateRP3,
    );
    expect(readSpy).toBeCalledWith(
      mockConnectedDevice.id,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.boudRateSerial,
    );
    expect(readSpy).toBeCalledWith(
      mockConnectedDevice.id,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.boudRateXD,
    );
  });

  it('should show error toast while fetching serial settings from BLE', async () => {
    jest
      .spyOn(BleManager, 'readCharacteristicForDevice')
      .mockImplementation(() => {
        throw new Error('Custom Exception');
      });
    const deviceType = 'MS-700';
    await store.dispatch(SerialSettingsFunction.getSerialSettings(deviceType));
    toastSpy = jest.spyOn(Toast, 'show');
    expect(toastSpy).toBeCalled();
  });

  it('should show the forwardingBehaviourModal when the forwardingBehaviourButton is pressed', () => {
    const {getByTestId, queryByTestId} = render(
      <NavigationContainer>
        <Serial navigation={navigation} />
      </NavigationContainer>,
    );
    const forwardingBehaviourButton = getByTestId('forwardingBehaviourButton');
    expect(forwardingBehaviourButton).toBeDefined();
    const forwardingBehaviourModal = queryByTestId('forwardingBehaviourModal');
    expect(forwardingBehaviourModal).toBeNull();
    fireEvent.press(forwardingBehaviourButton);
    expect(queryByTestId('forwardingBehaviourModal')).toBeDefined();
    const firstDropdownItem = getByTestId('forwardingBehaviourDropdownItem0');
    expect(firstDropdownItem).toBeDefined();
    const secondDropdownItem = getByTestId('forwardingBehaviourDropdownItem1');
    expect(secondDropdownItem).toBeDefined();
  });

  it('should save serial settings on BLE device', async () => {
    writeSpy = jest.spyOn(
      BleManager,
      'writeCharacteristicWithResponseForDevice',
    );
    jest.spyOn(base64, 'encode').mockImplementation(() => {
      return '1,2,3,4';
    });
    const deviceType = 'MS-700';
    const callbackFunction = () => {};
    await store.dispatch(
      SerialSettingsFunction.saveSerialSettings(deviceType, callbackFunction),
    );
    expect(writeSpy).toBeCalledWith(
      mockConnectedDevice.id,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.fwBehaviourRP1,
      '1,2,3,4',
    );
    expect(writeSpy).toBeCalledWith(
      mockConnectedDevice.id,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.fwBehaviourRP2,
      '1,2,3,4',
    );
    expect(writeSpy).toBeCalledWith(
      mockConnectedDevice.id,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.fwBehaviourRP3,
      '1,2,3,4',
    );
    expect(writeSpy).toBeCalledWith(
      mockConnectedDevice.id,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.fwBehaviourSerial,
      '1,2,3,4',
    );
    expect(writeSpy).toBeCalledWith(
      mockConnectedDevice.id,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.fwBehaviourTCP,
      '1,2,3,4',
    );
    expect(writeSpy).toBeCalledWith(
      mockConnectedDevice.id,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.boudRateXD,
      '1,2,3,4',
    );
    expect(writeSpy).toBeCalledWith(
      mockConnectedDevice.id,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.boudRateRP1,
      '1,2,3,4',
    );
    expect(writeSpy).toBeCalledWith(
      mockConnectedDevice.id,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.boudRateRP2,
      '1,2,3,4',
    );
    expect(writeSpy).toBeCalledWith(
      mockConnectedDevice.id,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.boudRateRP3,
      '1,2,3,4',
    );
    expect(writeSpy).toBeCalledWith(
      mockConnectedDevice.id,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.boudRateSerial,
      '1,2,3,4',
    );
    expect(writeSpy).toBeCalledWith(
      mockConnectedDevice.id,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.boudRateXD,
      '1,2,3,4',
    );
  });

  it('should show error toast while saving serial settings on BLE', async () => {
    jest
      .spyOn(BleManager, 'writeCharacteristicWithResponseForDevice')
      .mockImplementation(() => {
        throw new Error('Custom Exception');
      });
    await store.dispatch(SerialSettingsFunction.saveSerialSettings());
    toastSpy = jest.spyOn(Toast, 'show');
    expect(toastSpy).toBeCalled();
  });

  it('should fetch serial settings from CZA1300', async () => {
    readSpy = jest.spyOn(BleManager, 'readCharacteristicForDevice');
    const deviceType = 'CZA1300';
    await store.dispatch(SerialSettingsFunction.getSerialSettings(deviceType));
    expect(readSpy).toBeCalledWith(
      mockConnectedDevice.id,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.fwBehaviourRP1,
    );
    expect(readSpy).toBeCalledWith(
      mockConnectedDevice.id,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.fwBehaviourTCP,
    );
    expect(readSpy).toBeCalledWith(
      mockConnectedDevice.id,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.fwBehaviourXD,
    );
    expect(readSpy).toBeCalledWith(
      mockConnectedDevice.id,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.boudRateRP1,
    );
    expect(readSpy).toBeCalledWith(
      mockConnectedDevice.id,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.boudRateXD,
    );
  });

  it('should save serial settings on CZA1300', async () => {
    writeSpy = jest.spyOn(
      BleManager,
      'writeCharacteristicWithResponseForDevice',
    );
    jest.spyOn(base64, 'encode').mockImplementation(() => {
      return '1,2,3,4';
    });
    const deviceType = 'CZA1300';
    const callbackFunction = () => {};
    await store.dispatch(
      SerialSettingsFunction.saveSerialSettings(deviceType, callbackFunction),
    );
    expect(writeSpy).toBeCalledWith(
      mockConnectedDevice.id,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.fwBehaviourRP1,
      '1,2,3,4',
    );
    expect(writeSpy).toBeCalledWith(
      mockConnectedDevice.id,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.fwBehaviourTCP,
      '1,2,3,4',
    );
    expect(writeSpy).toBeCalledWith(
      mockConnectedDevice.id,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.boudRateXD,
      '1,2,3,4',
    );
    expect(writeSpy).toBeCalledWith(
      mockConnectedDevice.id,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.boudRateRP1,
      '1,2,3,4',
    );
    expect(writeSpy).toBeCalledWith(
      mockConnectedDevice.id,
      UUIDMappingMS700.rootServiceUDID,
      UUIDMappingMS700.boudRateXD,
      '1,2,3,4',
    );
  });
});
