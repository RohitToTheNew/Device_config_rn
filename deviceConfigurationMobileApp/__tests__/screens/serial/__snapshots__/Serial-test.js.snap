// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`on homescreen bottom tab mount should match the snapshot 1`] = `null`;
